wbooks/src/app/components/Header/index.tsx

```diff
@@ -1,9 +1,55 @@
 import React from 'react';
-import { Image } from 'react-native';
+import { View, ImageBackground, TextInput, Text, TouchableOpacity, Image, Platform } from 'react-native';
+import { useDispatch, useSelector } from 'react-redux';
+import actionCreators from '@redux/book/actions';
 import background from '@assets/General/bc_nav_bar.png';
+import icSearch from '@assets/General/ic_search_placeholder.png';
 
 import styles from './styles';
 
-const HeaderBackgroundImage = () => <Image source={background} style={styles.container} />;
+interface Props {
+  routeName?: any;
+}
 
-export default HeaderBackgroundImage;
+interface RootState {
+  book: { query: string };
+}
+
+const HeaderBackground = ({ routeName }: Props) => {
+  const dispatch = useDispatch();
+  const { query } = useSelector((state: RootState) => state.book);
+
+  const onChangeQuery = (text: string) => {
+    dispatch(actionCreators.setQuery(text));
+  };
+
+  const deleteQuery = () => {
+    dispatch(actionCreators.setQuery(''));
+  };
+
+  if (routeName === 'SearchFilter')
+    return (
+      <View style={Platform.OS === 'ios' && { ...styles.container }}>
```

Comment:
Because when I apply flex:1 in styles, the focus of textinput search crash in android. Additionally, iOS need it, without the flex: 1 the waves of header crash.

Reply:
Ok, I don't catch what "crash". But it's fine.